lms/static/scripts/frontend_apps/config.js

```diff
@@ -82,31 +90,37 @@ import { createContext } from 'preact';
  */
 
 /**
- * @typedef {'blackboard_missing_integration' |
- *  'canvas_invalid_scope' } OAuthErrorCode
- */
-
-/**
- * Configuration for the error dialog shown if authorizing access to an
- * OAuth API fails.
- *
- * @typedef OAuthErrorConfig
- * @prop {string|null} authUrl
- * @prop {string} errorDetails
- * @prop {OAuthErrorCode|null} errorCode
- * @prop {string[]} canvasScopes
+ *
+ * @typedef {'blackboard_missing_integration' | 'canvas_invalid_scope' | 'reused_consumer_key'} ConfigErrorCode
+ *
+ * Base interface for error information provided through this configuration object.
+ *
+ * @typedef ConfigErrorBase
+ * @prop {ConfigErrorCode} [errorCode]
+ * @prop {object|string} [errorDetails]
  */
 
 /**
- * @typedef {'reused_consumer_key'} ErrorCode
+ * Configuration for information describing general errors for use in
+ * `error-dialog` mode.
+ *
+ * @typedef ErrorDialogConfigBase
+ * @prop {Extract<'reused_consumer_key', ConfigErrorCode>} [errorCode]
+ *
+ * @typedef {ConfigErrorBase & ErrorDialogConfigBase} ErrorDialogConfig
  */
 
 /**
- * Configuration for general error dialogs
+ * Configuration for information describing errors specific to
+ * OAuth API fails for use in `oauth2-redirect-error` mode. This adds a couple
+ * of additional optional properties to the base ConfigErrorCode type.
+ *
+ * @typedef OAuthErrorConfigBase
+ * @prop {Extract<'blackboard_missing_integration' | 'canvas_invalid_scope', ConfigErrorCode>} [errorCode]
+ * @prop {string} [authUrl]
```

Comment:
Should all of these properties be optional?

Reply:
The back-end doesn't guarantee us an `errorCode`, so that is optional, yes, in all cases.
I don't know if there is any enforcement anywhere that if OAuthError stuff is present, `authUrl` should be provided. I'm not sure how to handle that.